spotinst_sdk2/clients/ocean/__init__.py

```diff
@@ -1693,3 +1694,662 @@ def list_migrations(self, ocean_id: str):
+        """
+        response = self.send_get(
+            url=self.__base_ocean_url + ocean_id + "/controllerHeartbeat",
+            entity_name="ocean (Cluster Heartbeat)"
+        )
+
+        formatted_response = self.convert_json(
+            response, self.camel_to_underscore)
+
+        return formatted_response["response"]["items"][0]
+
+    def create_ocean_cluster(self, ocean: gcp_ocean.Ocean):
+        """
+        Create an Ocean Cluster
+
+        # Arguments
+        ocean (Ocean): Ocean Object
+
+        # Returns
+        (Object): Ocean API response
+        """
+        ocean = gcp_ocean.OceanRequest(ocean)
+
+        excluded_missing_dict = self.exclude_missing(
+            json.loads(ocean.toJSON()))
+
+        formatted_missing_dict = self.convert_json(
+            excluded_missing_dict, self.underscore_to_camel)
+
+        body_json = json.dumps(formatted_missing_dict)
+
+        response = self.send_post(
+            body=body_json,
+            url=self.__base_ocean_cluster_url,
+            entity_name='ocean')
+
+        formatted_response = self.convert_json(response,
+                                               self.camel_to_underscore)
+
+        return formatted_response["response"]["items"][0]
+
+    def get_all_ocean_cluster(self):
+        """
+        List the configurations for all Ocean clusters in the specified account.
+
+        # Returns
+        (Object): Ocean API response
+        """
+
+        response = self.send_get(
+            url=self.__base_ocean_cluster_url,
+            entity_name="ocean"
+        )
+
+        formatted_response = self.convert_json(
+            response, self.camel_to_underscore)
+
+        return formatted_response["response"]["items"]
+
+    def delete_ocean_cluster(self, ocean_id: str):
+        """
+        Delete a specified Ocean cluster.
+
+        # Arguments
+        ocean_id (String): ID of the Ocean Cluster
+
+        # Returns
+        (Object): Ocean API response
+        """
+        return self.send_delete(
+            url=self.__base_ocean_cluster_url + "/" + ocean_id,
+            entity_name="ocean"
+        )
+
+    def get_ocean_cluster(self, ocean_id: str):
+        """
+        Get the configuration for a specified Ocean cluster.
+
+        # Arguments
+        ocean_id (String): ID of the Ocean Cluster
+
+        # Returns
+        (Object): Ocean API response
+        """
+        response = self.send_get(
+            url=self.__base_ocean_cluster_url + "/" + ocean_id,
+            entity_name="ocean"
+        )
+
+        formatted_response = self.convert_json(
+            response, self.camel_to_underscore)
+
+        return formatted_response["response"]["items"][0]
+
+    def update_ocean_cluster(self, ocean_id: str, ocean: gcp_ocean.Ocean):
+        """
+        All Ocean parameters are updatable, excluding the Name and controllerClusterId.
+        This API supports partial updates, so specific fields can be updated separately.
```

Comment:
please change this to Update an existing Ocean Cluster

Reply:
Done